{
  final ComputeException cause=Exceptions.findCause(e,ComputeException.class);
  if (cause != null) {
    throw cause;
  }
  LOG.error(e,e);
  final ComputeException exception=new ComputeException("InternalError",String.valueOf(e.getMessage()));
  if (Contexts.lookup().hasAdministrativePrivileges()) {
    exception.initCause(e);
  }
  throw exception;
}
